upload/src/addons/SV/TitleEditHistory/EditHistoryHandler/Thread.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace SV\TitleEditHistory\EditHistoryHandler;
+
+use XF\EditHistory\AbstractHandler;
+use XF\Mvc\Entity\Entity;
+
+class Thread extends AbstractHandler
+{
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 *
+	 * @return bool
+	 */
+	public function canViewHistory(Entity $content) {
+		return $content->canViewThreadTitleHistory() && $content->canView();
+	}
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 *
+	 * @return bool
+	 */
+	public function canRevertContent(Entity $content) {
+		return $content->canEdit();
+	}
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 *
+	 * @return string
+	 */
+	public function getContentTitle(Entity $content) {
+		$prefix = $content->getRelation('Prefix') ? "[".$content->getRelation('Prefix')->getTitle()."]" : "";
+		return $prefix . ' ' . $content->title;
+	}
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 *
+	 * @return string
+	 */
+	public function getContentText(Entity $content) {
+		return $content->title;
+	}
+
+	public function getContentLink(Entity $content) {
+		return \XF::app()->router('public')->buildLink('threads', $content);
+	}
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 *
+	 * @return array
+	 */
+	public function getBreadcrumbs(Entity $content) {
+		return $content->getBreadcrumbs();
+	}
+
+	/**
+	 * @param \SV\TitleEditHistory\XF\Entity\Thread|Entity $content
+	 * @param \XF\Entity\EditHistory $history
+	 * @param \XF\Entity\EditHistory|null $previous
+	 *
+	 * @return mixed
+	 */
+	public function revertToVersion(Entity $content, \XF\Entity\EditHistory $history, \XF\Entity\EditHistory $previous = null) {

```

Comment:
`revertToVersion` should `XF\Service\Thread\Editor` similar to how `\XF\EditHistory\Post` does